cli/cmd/inject.go

```diff
@@ -405,6 +433,13 @@ func injectResource(bytes []byte, options *injectOptions) ([]byte, error) {
 		return nil, err
 	}
 
+	// retrieve the `metadata/name` field for reporting later
+	var om objMeta
+	if err := yaml.Unmarshal(bytes, &om); err != nil {
+		return nil, err
+	}
+	report.name = fmt.Sprintf("%s/%s", k8s.ShortNameFromCanonicalResourceName(strings.ToLower(meta.Kind)), om.Name)
```

Comment:
`ShortNameFromCanonicalResourceName` only works for a subset of all Kubernetes object types, so if the inject yaml includes an object type that's not covered by that function, then we won't properly format the name. For instance, if we were trying to inject a Secret named "super", `report.name` would be "/super".

I don't think this is an issue in the current implementation, however, since we only print the names of objects that include pod specs in the report, and all objects that include pod specs are covered in `ShortNameFromCanonicalResourceName`. But I'm inclined to switch to using full names so that down the road we know that the name field is printable for all object types.

```
	report.name = fmt.Sprintf("%s/%s", strings.ToLower(meta.Kind), om.Name)
```